src/requests.ts

```diff
@@ -1,5 +1,23 @@
+import { KintoRequest, HttpMethod, KintoRecord } from "./types";
 import { createFormData } from "./utils";
 
+interface RequestOptions {
+  safe?: boolean;
+  headers?: Headers;
+  method?: HttpMethod;
+  gzipped?: boolean | null;
+  last_modified?: number;
+}
+type AddAttachmentRequestOptions = RequestOptions & {
+  last_modified?: number;
+  filename?: string;
+};
+
+type RequestBody = { data?: any; permissions?: Record<string, string[]> };
```

Comment:
I guess that `Record` is the `Map` type for typescript?

Reply:
Yes! Although after thinking about it a bit more I realized I wasn't using `Record` correctly here. I pushed an update that defines `permissions` as `Partial<Record<Permission, string[]>>`, along with a new `Permission` type that defines the allowed keys based on the API documentation. This allows for type-checking permission objects, ensuring that you don't pass an invalid permission. As I work through the rest of the library, I'll be able to refine the type in circumstances where it makes sense (for example, only allowing `read` and `write` on a record).